code/modules/power/supermatter/supermatter.dm

```diff
@@ -398,77 +591,173 @@
 
 	Consume(AM)
 
-
-/obj/machinery/power/supermatter_shard/proc/Consume(atom/movable/AM)
-	if(istype(AM, /mob/living))
+/obj/machinery/power/supermatter_crystal/proc/Consume(atom/movable/AM)
+	if(isliving(AM))
 		var/mob/living/user = AM
-		user.dust()
-		power += 200
-		message_admins("[src] has consumed [key_name_admin(user)] <A HREF='?_src_=holder;adminplayerobservecoodjump=1;X=[x];Y=[y];Z=[z]'>(JMP)</a>.")
+		if(user.status_flags & GODMODE)
+			return
+		message_admins("[src] has consumed [key_name_admin(user)] [ADMIN_JMP(src)].")
 		investigate_log("has consumed [key_name(user)].", "supermatter")
-	else if(isobj(AM) && !istype(AM, /obj/effect))
-		investigate_log("has consumed [AM].", "supermatter")
+		user.dust()
+		matter_power += 200
+	else if(istype(AM, /obj/singularity))
+		return
+	else if(istype(AM, /obj))
+		if(!istype(AM, /obj/effect))
+			var/suspicion = ""
+			if(AM.fingerprintslast)
+				suspicion = "last touched by [AM.fingerprintslast]"
+				message_admins("[src] has consumed [AM], [suspicion] [ADMIN_JMP(src)].")
+			investigate_log("has consumed [AM] - [suspicion].", "supermatter")
 		qdel(AM)
-
-	power += 200
-	supermatter_zap()
-
+	if(!istype(AM, /obj/effect))
+		matter_power += 200
 
 	//Some poor sod got eaten, go ahead and irradiate people nearby.
 	for(var/mob/living/L in range(10))
-		var/rads = 500 * sqrt( 1 / (get_dist(L, src) + 1) )
+		var/rads = 500 * sqrt( 1 / (get_dist(L, src) + 1))
```

Comment:
Can cause a divided by zero runtime if the mob in question is on the same tile as the SM. Might be possible if you open the crate when the mob is on/under the crate